packages/openapi/src/decorators.ts

```diff
@@ -159,9 +177,23 @@ export const $info: InfoDecorator = (
   }
   context.program.stateMap(infoKey).set(entity, data);
 };
-
+/** The extension name to be used to extend `@info`. */
+export const infoExtensionName = "infoExtension";
 export function getInfo(program: Program, entity: Namespace): AdditionalInfo | undefined {
-  return program.stateMap(infoKey).get(entity);
+  const info = program.stateMap(infoKey).get(entity);
+  const extensionsMap = getNamespaceExtensions(program, entity);
+  const extensions = extensionsMap.get(infoExtensionName);
+
+  /* The following info properties can be provided via extensions and should be overridden */
+  if (extensions && info) {
+    if (Object.prototype.hasOwnProperty.call(extensions, "contact")) {
+      info.contact = null;
+    }
+    if (Object.prototype.hasOwnProperty.call(extensions, "title")) {
+      info.title = null;
```

Comment:
As said extension on the namespace don’t belong in info either they belong at the root of the document. The solution to is is to have a SetIntoExtension function to hat update the info object only

Reply:
How about we set the extension on the namespace using the key `infoExtension`, since we don't have a type that can map to AdditionalInfo. Then in the `getInfoExtensions` we can use that key to retrieve the info extensions from the namespace extensions?

Reply:
Actually, never mind my question above. This method will be sufficient:

```Typescript
export function setInfo(program: Program, entity: Namespace, data: AdditionalInfo) {  
  program.stateMap(infoKey).set(entity, data);
}

```